prog/rand.go

```diff
@@ -119,10 +119,18 @@ func truncateToBitSize(v, bitSize uint64) uint64 {
 	return v & uint64(1<<bitSize-1)
 }
 
-func (r *randGen) randRangeInt(begin, end, bitSize uint64) uint64 {
+func (r *randGen) randRangeInt(begin, end, bitSize, align uint64) uint64 {
 	if r.oneOf(100) {
 		return r.randInt(bitSize)
 	}
+	if align != 0 {
+		if begin == 0 && int64(end) == -1 {
+			// Special [0:-1] range for all possible values.
+			end = uint64(math.Exp2(float64(bitSize)) - 1)
```

Comment:
Isn't it the same as `1<<bitSize - 1`? I am afraid of floating points.

Reply:
It is. Fixed with the last update.